activerecord/test/cases/dirty_test.rb

```diff
@@ -817,6 +817,40 @@ def catchphrase
     refute person.changed?
   end
 
+  test "saved_changes in double save" do
+    double_saved_changes = nil
+    klass = Class.new(ActiveRecord::Base) do
+      self.table_name = "people"
+
+      after_save do
+        if followers_count == 0
+          self.followers_count = 1
+          save!
+        else
+          double_saved_changes = saved_changes
+        end
+      end
+    end
+    person = klass.create!(first_name: "Sean")
+    assert_nil double_saved_changes["first_name"]
```

Comment:
Is the `double_saved_changes` variable necessary at all? Would this test be more clear if it just called `person.saved_changes` directly?

Reply:
That would be testing something different though, so wouldn't prevent a regression.  Within the nested save it is clear that `saved_changes` should return the changes corresponding to the save that caused the callback to be called.

In contrast, if you see code like

```ruby
    person = klass.create!(first_name: "Sean")
    changes = person.saved_changes
```

without context, then it would be more likely to assume that `saved_changes` refers to the call to `create!`.  That isn't the case and wasn't something I wanted to get into with this pull request.  However, without this PR it is currently returning an empty hash for `saved_changes`, which clearly doesn't make sense and is what is being tested in the following test in this PR.